Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialTypeAsset.cpp

```diff
@@ -76,6 +76,34 @@ namespace AZ
             AssetInitBus::Handler::BusDisconnect();
         }
 
+        bool MaterialTypeAsset::InitializeNonSerializedData()
+        {
+            if (m_isNonSerializedDataInitialized)
+            {
+                return true;
+            }
+            for (auto& shaderItem : m_generalShaderCollection)
+            {
+                if (!shaderItem.InitializeShaderOptionGroup())
+                {
+                    return false;
+                }
```

Comment:
For this and below, I just wanted to confirm that the desired behavior is to short-circuit and not call `InitializeShaderOptionGroup()` on any other shaders. Typically I'd expect you'd still want to call `InitializeShaderOptionGroup()` on all shaders, but return false if any one of them didn't return true.

Reply:
What you suggest is a good idea, but in reality if at least one fails, then it is a fatal error for the system anyways. Is like doing a lot of good work, but the crash will be imminent.